### ISSUE TITLE
sp_BlitzWho - Parallel blocked processes not showing a blocking_session_id

### ISSUE THREAD

**Issue by Adedba** [BrentOzarULTD/SQL-Server-First-Responder-Kit#2489]
**Version of the script**
SELECT @Version = '7.97', @VersionDate = '20200712';

**What is the current behavior?**
If a session running a parallel query is blocked it does not show as being blocked.

**If the current behavior is a bug, please provide the steps to reproduce.**

/* Query window 1 , Blocking query*/

BEGIN TRAN 
	UPDATE Users 
	SET Reputation = Reputation +1 
	WHERE Id < 100
--ROLLBACK 

/* Query window 2 , Parallel query blocked by Query window 1 */

SELECT Location,COUNT(*)
FROM Users 
GROUP BY Location

/* Query window 3, run sp_BlitzWho and obeserve the blocking_session_id column */
EXEC sp_BlitzWho;

The blocking_session_id column will show as NULL for Query window two, now if you stop query two and replace it with: 

(Single threaded query on my laptop) 

SELECT TOP 1 * 
FROM Users

run this and then run sp_BlitzWho again and you should see that the blocking_session_id column is now showing correctly.

**What is the expected behavior?**

blocking_session_id should not be NULL for the blocked session.

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
Tested on SQL2016

**Comment by Adedba** [comment]
Bit of a pain to narrow down this one, so here is what I did: 

Ran sp_BlitzWho with @Debug = 1,

Using the output I found the blocking_session_id column:

				   CASE WHEN r.blocking_session_id <> 0 AND blocked.session_id IS NULL 
							       THEN r.blocking_session_id
							       WHEN r.blocking_session_id <> 0 AND s.session_id <> blocked.blocking_session_id 
							       THEN blocked.blocking_session_id
							       ELSE NULL 
						      END AS blocking_session_id,
							  
Added some new columns alongside for debugging: 
						  
				   CASE WHEN r.blocking_session_id <> 0 AND blocked.session_id IS NULL 
							       THEN 1
							       WHEN r.blocking_session_id <> 0 AND s.session_id <> blocked.blocking_session_id 
							       THEN 2
							       ELSE NULL 
						      END AS blocking_clause,
							  r.blocking_session_id AS r_blocking_session_id,
							  blocked.session_id as blocked_session_id,
							  s.session_id AS s_session_id,
							  blocked.blocking_session_id AS blocked_blocking_session_id,
							  
This way I could see which clause in the case was being used, and all the blocking columns referenced so I could see the difference netween the parallel query and the single threaded query

![image](https://user-images.githubusercontent.com/33764798/88291097-a6729a80-ccef-11ea-89e8-0d9fc26a828b.png)


Here is where it gets interesting! r.blocking_session_id = 0 for parallel queries yet it shows the blocking session id if single threaded! so by adding a new clause to the CASE statement I think we can solve this.


Replace:
				   CASE WHEN r.blocking_session_id <> 0 AND blocked.session_id IS NULL 
							       THEN r.blocking_session_id
							       WHEN r.blocking_session_id <> 0 AND s.session_id <> blocked.blocking_session_id 
							       THEN blocked.blocking_session_id
							       ELSE NULL 
						      END AS blocking_session_id,
							  
							  
with: 

				   CASE WHEN r.blocking_session_id <> 0 AND blocked.session_id IS NULL 
							       THEN r.blocking_session_id
							       WHEN r.blocking_session_id <> 0 AND s.session_id <> blocked.blocking_session_id 
							       THEN blocked.blocking_session_id
								   WHEN r.blocking_session_id = 0 AND s.session_id = blocked.session_id 
								   THEN blocked.blocking_session_id
							       ELSE NULL 
						      END AS blocking_session_id,
							  
							  
I have tested with my scenario locally but this could probably use some more testing.

**Comment by Adedba** [comment]
Will do a commit now but won't submit a pull request just yet.

**Comment by BrentOzar** [comment]
Awesome, thanks sir! Will keep an eye out for the PR.

**Comment by Adedba** [comment]
Your welcome! On holiday this week so haven't had a chance to give it a thorough test, it fixed my scenario just wanted to be sure I haven't introduced other issues :) will pick it up again next week but let me know if you want the PR sooner as I can get that submitted if you wanted it sooner.

**Comment by BrentOzar** [comment]
Looks good, thanks sir! It's merged into dev and will be in the August release along with credit for you in the release notes.
